### PR TITLE
Use dep as the dependency manager

### PR DESCRIPTION
**Pull request by icyflame**
Makefile is a convenience feature - you can run commands like `make docker/build` after writing them once inside the Makefile. It's also very useful for new users who don't quite know where to begin.

### CODE REVIEW

**Comment by kshitij10496** [comment]
This is great!
I tried using `dep` for managing Go packages. However, it seemed to me that Heroku doesn't yet support it. Hence, I reverted to `godep`.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+docker/build:
+	docker build -t hercules:latest .
+
+docker/run_attached:
+	docker run --rm -p 8080:8080 hercules:latest
```
Comment (kshitij10496) [inline Makefile]: Should we set the port `8080` as the `PORT` environment variable to be used by the app in the Makefile or leave it in the Dockerfile?

Comment (icyflame) [reply Makefile]: The Makefile is mainly for development purposes. And considering that we would just be running the bundled Dockerfile and Makefile in development, it makes sense to just hardcode the ports here.

**Comment by icyflame** [comment]
@kshitij10496 Oh! Didn't know that. Feel free to just reject this PR in that case (been developing with Docker and Kubernetes lately, so I didn't consider Heroku support.

A cursory look at the Heroku page for support seems to show that `dep` is supported through the community. [here](https://devcenter.heroku.com/articles/go-support#supported-dependency-vendor-managers). Since this is a basic setup in this application, we shouldn't run into any problems. You can consider using it. 🙂

File: Makefile (hunk 1 above)
Comment (kshitij10496) [reply Makefile]: Sounds good!

Also, how do we connect the app running in the container to the local DB?
While testing the current Dockerfile for development, I couldn't get the DB connected.
Should the DB be run in a separate container and both of these containers interact to establish the connection?

P.S. I have read just about mounting volumes yet; haven't studied container-container interaction.

**Comment by kshitij10496** [comment]
> A cursory look at the Heroku page for support seems to show that dep is supported through the community. here. Since this is a basic setup in this application, we shouldn't run into any problems. You can consider using it. 

Awesome! Let's give `dep` a try. 🎉 
Let me know if this branch is ready for testing and deployment. 😄

**Comment by icyflame** [comment]
@kshitij10496 Cool! I will update my branch to the latest master and put these changes in and ping again. 🙂

**Comment by icyflame** [comment]
@kshitij10496 It's fine to merge now. I just did `dep ensure -v` after rebasing to the latest master.

**Comment by kshitij10496** [comment]
`dep` is live! 🕶 :party_parrot:

I just had to add some metadata in [`Gopkg.toml`](https://github.com/kshitij10496/hercules/blob/77982c8ba231b6ad8f49597754b5db478218f1fa/Gopkg.toml#L27-L31) in order to get it working.

Refer: [Build Configuration](https://devcenter.heroku.com/articles/go-apps-with-dep#build-configuration)

**Pull request merged**